src/core/server/http/cookie_session_storage.test.ts

```diff
@@ -424,4 +424,60 @@ describe('Cookie based SessionStorage', () => {
       ]);
     });
   });
+
+  describe('#options', () => {
+    describe('#SameSite', () => {
+      it('throws an exception if "SameSite: None" set on not Secure connection', async () => {
+        const { server: innerServer } = await server.setup(setupDeps);
+
+        expect(
+          createCookieSessionStorageFactory(logger.get(), innerServer, {
+            ...cookieOptions,
+            sameSite: 'None',
+          })
+        ).rejects.toThrowErrorMatchingInlineSnapshot(
+          `"\\"SameSite: None\\" requires Secure connection"`
+        );
+      });
+
+      for (const sameSite of ['Strict', 'Lax', 'None'] as ['Strict', 'Lax', 'None']) {
```

Comment:
super nit: I _think_ you can do this instead:
```suggestion
      for (const sameSite of ['Strict', 'Lax', 'None'] as const) {
```